### PR TITLE
[makeOptions] Add new enum from swift Options

### PR DESCRIPTION
**Pull request by cachemeifyoucan**
Add new flag options handling for newly added Flags in swift repo. Also updating the Options to match.

### CODE REVIEW

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

**Review by artemcm — APPROVED** [review]

File: Sources/SwiftOptions/Options.swift [hunk 1]
```diff
@@ -251,7 +251,7 @@ extension Option {
...
   public static let driverVerifyFineGrainedDependencyGraphAfterEveryImport: Option = Option("-driver-verify-fine-grained-dependency-graph-after-every-import", .flag, attributes: [.helpHidden, .doesNotAffectIncrementalBuild], helpText: "Debug DriverGraph by verifying it after every import", group: .internalDebug)
   public static let driverWarnUnusedOptions: Option = Option("-driver-warn-unused-options", .flag, attributes: [.helpHidden], helpText: "Emit warnings for any provided options which are unused by the driver")
-  public static let dumpApiPath: Option = Option("-dump-api-path", .separate, attributes: [.helpHidden, .frontend, .noDriver], helpText: "The path to output swift interface files for the compiled source files")
+  public static let dumpApiPath: Option = Option("-dump-api-path", .separate, attributes: [.frontend, .noDriver, .cacheInvariant], helpText: "The path to output swift interface files for the compiled source files")
```
Comment (tshortli) [inline Sources/SwiftOptions/Options.swift:254]: Is the removal of `.helpHidden` intentional?

**Review by tshortli — APPROVED** [review]
Thanks for the clean up!

File: Sources/SwiftOptions/Options.swift (hunk 1 above)
Comment (cachemeifyoucan) [reply Sources/SwiftOptions/Options.swift:254]: It is probably unintentional in the swift options.td change.

Grouping is kind of hard in options.td, maybe we should take the visibility field in upstream llvm so we can remove all the driver tool related flags into visibility field.

**Pull request merged**